modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/package.scala

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.giapi
+
+import cats.implicits._
+import cats.effect._
+import edu.gemini.aspen.giapi.util.jms.status.StatusGetter
+import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
+import fs2.async
+
+package object client {
+
+  /**
+    * Allowed types according to GIAPI
+    */
+  implicit val strItemGetter: ItemGetter[String]    = new ItemGetter[String] {}
+  implicit val doubleItemGetter: ItemGetter[Double] = new ItemGetter[Double] {}
+  implicit val intItemGetter: ItemGetter[Int]       = new ItemGetter[Int]    {}
+  implicit val floatItemGetter: ItemGetter[Float]   = new ItemGetter[Float]  {}
+}
+
+package client {
+
+  /**
+    * Typeclass to present as evidence when calling `Giapi.get`
+    */
+  sealed trait ItemGetter[A]
+
+  /**
+    * Represents a connection to a GIAPi based instrument
+    *
+    * @tparam F Effect type
+    */
+  trait GiapiConnection[F[_]] {
+    def connect: F[Giapi[F]]
+  }
+
+  /**
+    * Algebra to interact with a GIAPI instrument
+    *
+    * @tparam F Effect Type
+    */
+  trait Giapi[F[_]] {
+
+    /**
+      * Returns a value for the status item. If not found or there is an error, e.g. on types the exception is returned
+      */
+    def get[A: ItemGetter](statusItem: String): F[Option[A]]
```

Comment:
@tpolecat I changed this with a typeclass and it works nicely. I'm wondering if I should return `F[Either[Throwable, A]]` rather than eating the exception and converting it to `None`. Any opinions?

Reply:
The "right" way to do this in MTL style is

```scala
def get[A: ItemGetter](statusItem: String)(
  implicit ev: ApplicativeError[F, String]
): F[A]
```

which would require that `F` provide a way to return an error as a string. A concrete `F` that would allow this is `EitherT[IO, String, ?]` which as you know is just a wrapper for `IO[Either[String, ?]]`, which is what you were thinking in the first place.  The advantage here is that you don't have to talk about the error case in your business logic unless you really want to, in which case you can use `.attempt` to turn an `F[A]` into an `F[Either[String, A]]`.

Reply:
If I try to do this, I end up with errors defining my `GPIClient` in terms of `F` as I can't produce a generic `ApplicativeError[F, E]` or even if I do it with `String` there isn't a generic `ApplicativeError[F, String]`

Am i missing something?

Reply:
> I can't produce a generic ApplicativeError[F, E] or even if I do it with String there isn't a generic ApplicativeError[F, String]

I'm not sure what you mean here. `ev` will be available if `F` can treat `String` as an error, like `Either[String, ?]` for instance. So you demand it anywhere you need to raise or trap an error. I'm just waking up but I'll be around in a couple hours if you want to chat.